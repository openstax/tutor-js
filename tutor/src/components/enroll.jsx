import PropTypes from 'prop-types';
import React from 'react';
import { observer } from 'mobx-react';
import { Modal } from 'react-bootstrap';
import Enroll from '../models/course/enroll';
import Router from '../helpers/router';
import { withRouter } from 'react-router-dom';
import studentIDForm from './enroll/student-id';
import invalidCode from './enroll/invalid-code';
import invalidLinks from './enroll/invalid-links-use';
import invalidLMS from './enroll/invalid-lms-use';
import invalidTeacher from './enroll/invalid-teacher';
import selectPeriod from './enroll/select-periods';
import droppedStudent from './enroll/dropped-student';
import courseEnded from './enroll/course-ended';
import unknownError from './enroll/unknown-error';

@withRouter
@observer
export default
class CourseEnroll extends React.Component {

  static propTypes = {
      enrollment: PropTypes.object,
      history: PropTypes.object.isRequired,
  }

  static Components = {
      studentIDForm,
      selectPeriod,
      invalidLMS,
      invalidLinks,
      invalidCode,
      invalidTeacher,
      droppedStudent,
      courseEnded,
      unknownError,
  }

  enrollmentCode = Router.currentParams().enrollmentCode;
  enrollment = this.props.enrollment ||
    new Enroll({ enrollment_code: this.enrollmentCode, history: this.props.history });

  UNSAFE_componentWillMount() {
      this.enrollment.create();
  }

  render() {
      const { enrollment } = this;
      if (!enrollment.bodyContents) {
          return null;
      }

      return (
          <Modal.Dialog
              className="course-enroll"
          >
              {enrollment.bodyContents}
          </Modal.Dialog>
      );
  }
}
