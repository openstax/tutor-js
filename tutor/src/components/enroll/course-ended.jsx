import React from 'react';
import { Modal } from 'react-bootstrap';
import { Icon } from 'shared';

export default function droppedStudent() {
    return (
        <div>
            <Modal.Header className="warning">
                <Icon type="exclamation-triangle" />
                <span>Sorry, the course has ended</span>
            </Modal.Header>
            <Modal.Body>
                <p>
          This is an old course enrollment link. Ask your instructor for the new link.
                </p>
            </Modal.Body>
        </div>
    );
}
