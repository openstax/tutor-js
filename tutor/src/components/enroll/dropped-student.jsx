import React from 'react';
import { Modal } from 'react-bootstrap';
import { Icon } from 'shared';

export default function droppedStudent() {
    return (
        <div>
            <Modal.Header className="warning">
                <Icon type="exclamation-triangle" />
                <span>Sorry, you have been removed from this course</span>
            </Modal.Header>
            <Modal.Body>
        Please contact your instructor if you need access to this course.
            </Modal.Body>
        </div>
    );
}
