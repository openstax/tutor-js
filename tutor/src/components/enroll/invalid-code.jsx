import React from 'react';
import { Modal, Alert } from 'react-bootstrap';

export default function invalidCode() {
    return (
        <Modal.Body>
            <Alert variant="danger">
                The provided enrollment code is not valid. Please verify the enrollment code and try again.
            </Alert>
        </Modal.Body>
    );
}
