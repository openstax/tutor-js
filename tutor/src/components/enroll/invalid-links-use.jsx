import React from 'react';
import { Modal } from 'react-bootstrap';
import { Icon } from 'shared';

export default function invalidLinksUse() {
    return (
        <div>
            <Modal.Header className="warning">
                <Icon type="exclamation-triangle" />
                <span>Sorry, this enrollment link isnâ€™t valid</span>
            </Modal.Header>
            <Modal.Body>
                To enroll in this OpenStax Tutor course, access your course in your
                institution's Learning Management System (LMS). Launch OpenStax Tutor
                from your LMS.
            </Modal.Body>
        </div>
    );
}
