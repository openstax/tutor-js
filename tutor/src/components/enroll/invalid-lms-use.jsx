import React from 'react';
import { Modal } from 'react-bootstrap';
import { Icon } from 'shared';

export default function invalidLmsUse() {
    return (
        <div>
            <Modal.Header className="warning">
                <Icon type="exclamation-triangle" />
                <span>Sorry, you need an enrollment link</span>
            </Modal.Header>
            <Modal.Body>
        To enroll in this OpenStax Tutor course, use the course enrollment
        link provided by your instructor.
            </Modal.Body>
        </div>
    );
}
