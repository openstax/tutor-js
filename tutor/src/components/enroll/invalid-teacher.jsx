import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import UserMenu from '../../models/user/menu';
import PropTypes from 'prop-types';

export default function InvalidTeacher({ enrollment }) {

  return (
    <div className="is-teacher">
      <Modal.Body>
        <h3>Sorry, you can’t enroll as a student in your course using this email address.</h3>
        <p>
          It looks like you’re trying to enroll in {enrollment.courseName} using your instructor account.
        </p>
        <p>
          Contact <a href={`mailto:${UserMenu.supportEmail}`}>Support</a> if you need help.
        </p>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="primary" className="cancel" onClick={enrollment.onCancel}>
          Continue to My courses
        </Button>
      </Modal.Footer>
    </div>
  );
}

InvalidTeacher.propTypes = {
  enrollment: PropTypes.shape({
    courseName: PropTypes.string,
    onCancel: PropTypes.func,
  }),
}
