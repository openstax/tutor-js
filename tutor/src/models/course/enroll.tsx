import React from 'react';
import {
    BaseModel,
    field,
    computed,
    action,
    when,
    observable,
    modelize,
    NEW_ID,
} from 'shared/model';
import { get, pick, isEmpty } from 'lodash';
import { Redirect } from 'react-router-dom';
import Courses, { Course } from '../courses-map';
import S from '../../helpers/string';
import Router from '../../../src/helpers/router';
import Activity from 'shared/components/staxly-animation';
import Enroll from '../../../src/components/enroll';
import User from '../user';

export default class CourseEnrollment extends BaseModel {

    @field id = NEW_ID;
    @field enrollment_code;

    @field student_identifier;
    @observable pendingEnrollmentCode;
    @observable originalEnrollmentCode;
    @observable to: any = {};
    @observable status;
    @observable history?: any;
    @observable isComplete = false;
    @observable courseToJoin;
    @observable isLoadingCourses;
    @observable courses = Courses;

    constructor(attrs = {}) {
        super(attrs);
        modelize(this);
        if (attrs.courses) { this.courses = attrs.courses; }
        this.originalEnrollmentCode = this.enrollment_code;
        when(
            () => this.isRegistered,
            () => this.fetchCourses(),
        );
    }

    @computed get bodyContents() {
        if (User.shouldSignTerms) {
            return null;
        }

        if (this.isLoading) {
            return <Activity isLoading={true} />;
        } else if (this.isFromLms && false === this.courseIsLmsEnabled) {
            return this.renderComponent('invalidLMS');
        } else if (!this.isFromLms && true === this.courseIsLmsEnabled) {
            return this.renderComponent('invalidLinks');
        } else if (this.isTeacher) {
            return this.renderComponent('invalidTeacher');
        } else if (this.isInvalid) {
            return this.renderComponent('invalidCode');
        } else if (this.isComplete) {
            if (this.courseId) {
                return <Redirect to={Router.makePathname('dashboard', { courseId: this.courseId })} />;
            } else {
                // the BE says we're registered but we didn't find the course
                // most likely because they're coming from LMS or have a non-standard enrollment
                return <Redirect to={Router.makePathname('myCourses')} />;
            }
        } else if (!isEmpty(this.api.errors)) {
            if (this.api.errors.dropped_student) {
                return this.renderComponent('droppedStudent');
            } else if (this.api.errors.course_ended) {
                return this.renderComponent('courseEnded');
            } else {
                return this.renderComponent('unknownError');
            }
        } else if (this.needsPeriodSelection) {
            return this.renderComponent('selectPeriod');
        } else {
            return this.renderComponent('studentIDForm');
        }
    }

    renderComponent(name) {
        const Tag = Enroll.Components[name];
        return <Tag enrollment={this} errors={this.api.errors} />;
    }

    @action.bound
    selectPeriod(period) {
        this.pendingEnrollmentCode = period.enrollment_code;
    }

    @computed get periodIsSelected() {
        return Boolean(this.pendingEnrollmentCode);
    }

    @action.bound
    onSubmitPeriod() {
        this.courseToJoin = null;
        this.enrollment_code = this.pendingEnrollmentCode;
        this.create();
    }

    @computed get courseIsLmsEnabled() {
        return this.courseToJoin ?
            this.courseToJoin.is_lms_enabled : get(this, 'to.course.is_lms_enabled', null);
    }

    @action.bound
    onCancel() {
        this.history.push(Router.makePathname('myCourses'));
    }

    @action.bound
    onCancelStudentId() {
        this.student_identifier = '';
        this.confirm();
    }

    @action.bound
    onStudentIdContinue() {
        this.confirm();
    }

    @computed get courseName() {
        if (this.isTeacher) {
            return get(this.api.errors, 'is_teacher.data.course_name', '');
        }
        return get(this, 'to.course.name', '');
    }

    @computed get course() {
        return this.courses.array.find(c =>
            c.periods.find(p =>
                p.enrollment_code == this.enrollment_code
            )
        );
    }

    @computed get courseId() {
        return get(this, 'to.course.id', this.course ? this.course.id : '');
    }

    @computed get isPending() {
        return Boolean(isEmpty(this.api.errors) && isEmpty(this.to));
    }

    @computed get isInvalid() {
        return Boolean(this.api.errors && this.api.errors.invalid_enrollment_code);
    }

    @computed get isTeacher() {
        return Boolean(get(this.api.errors, 'is_teacher'));
    }

    @computed get isRegistered() {
        return Boolean(get(this.api.errors, 'already_enrolled') || this.status == 'processed');
    }

    @computed get isLoading() {
        return Boolean(this.api.isPending || this.isLoadingCourses);
    }

    @computed get needsPeriodSelection() {
        return Boolean(this.isFromLms && S.isUUID(this.enrollment_code));
    }

    @computed get isFromLms() {
        return S.isUUID(this.originalEnrollmentCode);
    }

    fetchCourses() {
        this.isLoadingCourses = true;
        this.courses.fetch().then(() => {
            if (this.course) { // should always be set but maybe the BE messes up and doesn't return the new course yet?
                this.course.studentTaskPlans.expecting_assignments_count = get(this, 'to.period.assignments_count', 0);
            }
            this.isLoadingCourses = false;
            this.isComplete = true;
            // enrolling in a course may trigger new terms
            User.terms.fetch();
        });
    }

    // called by api
    @action create() {
        if (this.needsPeriodSelection) {
            this.courseToJoin = new Course();
            return { method: 'GET', url: `enrollment/${this.originalEnrollmentCode}/choices` };
        }
        return { data: pick(this, 'enrollment_code') };
    }

    @action onEnrollmentCreate({ data }) {

        if (this.needsPeriodSelection) {
            const c = new Course(data);
            this.courseToJoin = c;
            if (this.courseToJoin.periods.length == 1) {
                this.enrollment_code = this.courseToJoin.periods[0].enrollment_code;
                this.create();
            }
        } else {
            this.update(data);
        }
    }

    confirm() {
        return { data: pick(this, 'student_identifier') };
    }
}
