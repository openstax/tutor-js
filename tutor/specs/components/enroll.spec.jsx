import { R, Factory, TutorRouter, hydrateModel, runInAction } from '../helpers';
import Enroll from '../../src/components/enroll';
import { CourseEnrollment } from '../../src/helpers/enroll';

jest.mock('../../src/helpers/router');
jest.mock('../../src/models/user', () => ({
    terms: { fetch: jest.fn() },
}));

describe('Student Enrollment', () => {
    let params;
    let enrollment;
    let coursesMap;
    let fetchMock;
    let course;

    beforeEach(() => {
        coursesMap = Factory.coursesMap();
        fetchMock = Promise.resolve();
        coursesMap.fetch = jest.fn(() => fetchMock);
        course = coursesMap.array[0];
        coursesMap.set(course.id, course);
        params = { courseId: course.id };

        enrollment = hydrateModel(CourseEnrollment, {
            history: { push: jest.fn() },
            courses: coursesMap, enrollment_code: '1234',
        });
        enrollment.create = jest.fn();
        TutorRouter.currentParams.mockReturnValue(params);
        TutorRouter.makePathname = jest.fn((name) => name);
    });

    it('loads when mounted', async () => {
        enrollment.isLoadingCourses = true;
        const wrapper = mount(<R><Enroll enrollment={enrollment} /></R>);
        expect(await axe(wrapper.html())).toHaveNoViolations();
        expect(enrollment.create).toHaveBeenCalled();
        expect(wrapper).toHaveRendered('StaxlyAnimation');
        expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
    });


    it('blocks teacher enrollment', () => {
        enrollment.api.errors = { is_teacher: { data: { course_name: 'My Fairly Graded Course' } } };
        expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
        const enroll = mount(<R><Enroll enrollment={enrollment} /></R>);

        TutorRouter.makePathname = jest.fn(() => '/courses');
        enroll.find('Button').simulate('click');
        expect(TutorRouter.makePathname).toHaveBeenCalledWith('myCourses');
        expect(enrollment.history.push).toHaveBeenCalledWith('/courses');
    });

    it('forwards to your course if already a member', (done) => {
        enrollment.api.errors = { already_enrolled: { data: { course_name: 'My Course' } } };
        const enroll = mount(<R><Enroll enrollment={enrollment} /></R>);
        expect(enroll).toHaveRendered('StaxlyAnimation');
        enrollment.isComplete = true;
        enroll.update();
        setTimeout(() => {
            expect(enroll).toHaveRendered('Redirect');
            enroll.unmount();
            done();
        }, 100);
    });

    it('displays an invalid message', () => {
        enrollment.api.errors = { invalid_enrollment_code: true };
        expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
    });

    it('displays generic error message', () => {
        enrollment.api.errors = [{ code: 'blah', message: 'this is a error that we cant handle' }];
        expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
    });

    describe('select periods', () => {
        beforeEach(() => {
            runInAction(() =>
                enrollment.enrollment_code = enrollment.originalEnrollmentCode = 'cc3c6ff9-83d8-4375-94be-8c7ae3024938'
            );


            enrollment.onEnrollmentCreate({
                data: {
                    name: 'My Grand Course',

                    periods: [
                        { name: 'Period #1', enrollment_code: '1234' },
                        {
                            name: 'Period #2',
                            enrollment_code: '4321',
                        },
                    ],
                },
            });
        });

        it('matches snapshot', () => {
            expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
        });

        it('can display a list of periods when joining from enrollment launch uuid', () => {
            const enroll = mount(<R><Enroll enrollment={enrollment} /></R>);
            expect(enroll).toHaveRendered('SelectPeriod');
            enroll.find('.choice').last().simulate('click');
            enroll.find('.btn-primary').simulate('click');
            expect(enrollment.pendingEnrollmentCode).toEqual('4321');

        });

        it('skips period selection when course has only one', () => {
            enrollment.create = jest.fn();
            enrollment.onEnrollmentCreate({
                data: {
                    name: 'My Grand Course', periods: [{
                        name: 'single period', enrollment_code: '4321',
                    }],
                },
            });
            const enroll = mount(<R><Enroll enrollment={enrollment} /></R>);
            expect(enroll).toHaveRendered('StudentIDForm');
        });
    });

    it('submits with student id when form is clicked', () => {
        const enroll = mount(<R><Enroll enrollment={enrollment}  /></R>);
        expect(enroll).toHaveRendered('StudentIDForm');
        expect.snapshot(<R><Enroll enrollment={enrollment} /></R>).toMatchSnapshot();
        enroll.find('input').simulate('change', { target: { value: '411' } });
        expect(enrollment.student_identifier).toEqual('411');
        enrollment.confirm = jest.fn();
        enroll.find('.btn-primary').simulate('click');
        expect(enrollment.confirm).toHaveBeenCalled();
    });

    it('redirects on success', () => {
        enrollment.to = { course: { id: '133' } };
        enrollment.isComplete = true;
        const enroll = mount(<R><Enroll enrollment={enrollment}  /></R>);
        expect(enroll).toHaveRendered('Redirect');
    });

});
