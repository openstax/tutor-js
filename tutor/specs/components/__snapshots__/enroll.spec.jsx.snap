// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Student Enrollment blocks teacher enrollment 1`] = `
<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div
      className="is-teacher"
    >
      <div
        className="modal-body"
      >
        <h3>
          Sorry, you can’t enroll as a student in your course using this email address.
        </h3>
        <p>
          It looks like you’re trying to enroll in 
          My Fairly Graded Course
           using your instructor account.
        </p>
        <p>
          Contact 
          <a
            href="mailto:support@openstax.org"
          >
            Support
          </a>
           if you need help.
        </p>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="cancel btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Continue to My courses
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Enrollment displays an invalid message 1`] = `
<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-body"
    >
      <div
        className="fade alert alert-danger show"
        role="alert"
      >
        The provided enrollment code is not valid. Please verify the enrollment code and try again.
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Enrollment displays generic error message 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div>
      <div
        className="warning modal-header"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-exclamation-triangle fa-w-18 ox-icon ox-icon-exclamation-triangle c0"
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span>
          Sorry, an error occured
        </span>
      </div>
      <div
        className="modal-body"
      >
        <p>
          blah: this is a error that we cant handle
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Enrollment loads when mounted 1`] = `
.c0 {
  margin: auto;
  padding: 20px;
}

.c1 {
  display: block;
  height: 100%;
  margin: 20px auto 0 auto;
  overflow: visible;
  position: relative;
  padding-top: 20px;
  max-width: 200px;
}

.c6 {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
  fill: #77AF42;
  -webkit-animation: hkCfKg 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: hkCfKg 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center left;
  -ms-transform-origin: center left;
  transform-origin: center left;
}

.c5 {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
  fill: #F47641;
  -webkit-animation: kSacDd 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: kSacDd 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c4 {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
  fill: #5E6062;
  -webkit-animation: fXlVAV 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: fXlVAV 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c3 {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
  fill: #F4D019;
  -webkit-animation: hbytGi 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.6s infinite alternate;
  animation: hbytGi 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.6s infinite alternate;
  -webkit-transform-origin: center right;
  -ms-transform-origin: center right;
  transform-origin: center right;
}

.c2 {
  -webkit-animation-delay: 0.5s;
  animation-delay: 0.5s;
  fill: #00246A;
  -webkit-animation: jfsUaa 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  animation: jfsUaa 0.6s cubic-bezier(0.81,0.41,0.13,0.74) 0.5s infinite alternate;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div
      className="staxly-animation"
    >
      <div
        className="overlay"
      />
      <div
        className="inner c0"
      >
        <svg
          className="c1"
          enableBackground="new 0 -25 57.6 66"
          id="Layer_1"
          version="1.1"
          viewBox="0 -25 57.6 66"
          x="0px"
          xmlns="http://www.w3.org/2000/svg"
          y="0px"
        >
          <path
            className="blue c2"
            d="M38.4,22c0,0.4-0.3,0.7-0.7,0.7H13.5c-0.4,0-0.7-0.3-0.7-0.7v-1.4c0-0.4,0.3-0.7,0.7-0.7h24.2     c0.4,0,0.7,0.3,0.7,0.7V22z"
          />
          <path
            className="yellow c3"
            d="M41.8,17.8c0,0.4-0.3,0.7-0.7,0.7l-25.4,0.8c-0.4,0-0.7-0.3-0.7-0.7l-0.1-2.1     c0-0.4,0.3-0.7,0.7-0.7L41,15c0.4,0,0.7,0.3,0.7,0.7L41.8,17.8z"
          />
          <path
            className="gray c4"
            d="M34.8,14c0,0.4-0.3,0.7-0.7,0.7H11.6c-0.4,0-0.7-0.3-0.7-0.7v-3.5c0-0.4,0.3-0.7,0.7-0.7h22.5     c0.4,0,0.7,0.3,0.7,0.7V14z"
          />
          <path
            className="orange c5"
            d="M39,8.1c0,0.3-0.4,0.5-1,0.5H8.6c-0.5,0-1-0.2-1-0.5V6.9c0-0.3,0.4-0.5,1-0.5H38     c0.5,0,1,0.2,1,0.5V8.1z"
          />
          <path
            className="green c6"
            d="M43.9,5c0,0.7-0.6,1.2-1.2,1.1L15,4.8c-0.7,0-1.2-0.6-1.1-1.2L14,1.1c0-0.7,0.6-1.2,1.2-1.1     l27.7,1.3c0.7,0,1.2,0.6,1.1,1.2L43.9,5z"
          />
          <path
            className="st2"
            d="M3.1,30.2c-1.8,0-3.1,1.3-3.1,3.1c0,1.8,1.3,3.1,3.1,3.1c1.8,0,3.1-1.3,3.1-3.1     C6.2,31.5,4.9,30.2,3.1,30.2z M3.1,35.7c-1.4,0-2.3-1.1-2.3-2.4c0-1.3,0.9-2.4,2.3-2.4c1.4,0,2.3,1.1,2.3,2.4     C5.4,34.6,4.5,35.7,3.1,35.7z M10.9,30.2c-0.9,0-1.9,0.4-2.4,1.2h0v-1H7.7v8.8h0.8v-3.9h0C9,36,10,36.4,10.9,36.4     c1.8,0,3.1-1.3,3.1-3.1C14,31.5,12.7,30.2,10.9,30.2z M10.9,35.7c-1.4,0-2.5-1.1-2.5-2.4c0-1.3,1.1-2.4,2.5-2.4     c1.4,0,2.3,1.1,2.3,2.4C13.2,34.6,12.3,35.7,10.9,35.7z M18.3,30.2c-1.7,0-2.9,1.3-2.9,3.1c0,1.8,1.2,3.1,3,3.1c1,0,2-0.4,2.6-1.3     l-0.6-0.5c-0.4,0.6-1.2,1-2,1c-1.8,0-2.2-1.6-2.2-2.2h5v-0.5C21.1,31.6,20.1,30.2,18.3,30.2z M16.1,32.8c0-0.3,0.5-1.9,2.1-1.9     c1.1,0,2,0.9,2,1.9H16.1z M27.9,32.7v3.5h-0.8v-3.5c0-1-0.3-1.9-1.5-1.9c-1.1,0-2,0.8-2,2.4v2.9h-0.8V32c0-0.4-0.1-1.4-0.1-1.7h0.8     c0,0.4,0,0.9,0.1,1h0c0.3-0.7,1.1-1.2,2-1.2C27.4,30.2,27.9,31.4,27.9,32.7z M34.2,34.4c0,1.5-1.4,2-2.6,2c-0.9,0-1.8-0.2-2.4-0.9     l1-1c0.4,0.4,0.8,0.8,1.5,0.8c0.4,0,1-0.2,1-0.7c0-1.3-3.3-0.3-3.3-2.6c0-1.4,1.2-2,2.5-2c0.8,0,1.7,0.3,2.2,0.9l-1,0.9     c-0.3-0.4-0.7-0.6-1.2-0.6c-0.4,0-0.9,0.2-0.9,0.7C30.9,32.9,34.2,32,34.2,34.4z M37.5,30.1h1.7v1.3h-1.7v2.7c0,0.6,0.2,1,0.9,1     c0.3,0,0.6-0.1,0.8-0.2v1.3c-0.3,0.2-0.9,0.2-1.2,0.2c-1.6,0-2-0.7-2-2.2v-2.8h-1.3v-1.3H36v-1.8h1.5V30.1z M43,30     c-1,0-1.9,0.3-2.6,1l0.8,0.8c0.4-0.4,1-0.6,1.6-0.6c0.8,0,1.4,0.4,1.4,1.1v0.2h-0.4c-1.5,0-3.9,0.1-3.9,2.1c0,1.2,1.1,1.8,2.2,1.8     c0.8,0,1.5-0.3,2-1h0v0.8h1.4v-3.5C45.6,31.8,45.6,30,43,30z M44.1,33.8c0,0.9-0.5,1.4-1.5,1.4c-0.5,0-1.1-0.2-1.1-0.8     c0-0.9,1.5-0.9,2.3-0.9h0.3V33.8z M50.8,32.9l2.5,3.3h-1.9L49.8,34l-1.6,2.2h-1.8l2.5-3.3l-2.1-2.8h1.9l1.2,1.8l1.3-1.8h1.7     L50.8,32.9z M57.6,30.1v1.7h-0.3v-1.5h0l-0.6,1.5h-0.2l-0.6-1.5h0v1.5h-0.3v-1.7h0.4l0.5,1.3l0.5-1.3H57.6z M53.9,30.1h1.4v0.3h-0.5     v1.5h-0.3v-1.5h-0.5V30.1z"
            fill="#5E6062"
            id="ox-type"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Enrollment select periods matches snapshot 1`] = `
.c0 {
  margin-right: 0.5rem;
  margin-left: 0.5rem;
}

<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div
      className="enroll-form periods"
    >
      <div
        className="modal-body"
      >
        <div
          className="title"
        >
          <p
            className="joining"
          >
            You are joining
          </p>
          <h4>
            My Grand Course
          </h4>
        </div>
        <p>
          Please select the OpenStax Tutor section you are a member of. If you don't know which section to select, ask your instructor.
        </p>
        <div
          className="choices-listing list-group"
          onKeyDown={[Function]}
        >
          <div
            aria-pressed={false}
            className="list-group-item choice"
            onClick={[Function]}
            role="button"
          >
            <div
              className="content"
            >
              Period #1
            </div>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 ox-icon ox-icon-circle c0"
              color="#5e5e5e"
              data-icon="circle"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
          <div
            aria-pressed={false}
            className="list-group-item choice"
            onClick={[Function]}
            role="button"
          >
            <div
              className="content"
            >
              Period #2
            </div>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-circle fa-w-16 ox-icon ox-icon-circle c0"
              color="#5e5e5e"
              data-icon="circle"
              data-prefix="far"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-success btn btn-primary"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Continue
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Student Enrollment submits with student id when form is clicked 1`] = `
<div
  className="modal-dialog course-enroll"
>
  <div
    className="modal-content"
  >
    <div
      className="enroll-form"
    >
      <div
        className="modal-body"
      >
        <div
          className="title"
        >
          <p
            className="joining"
          >
            You are joining
          </p>
          <h4>
            
          </h4>
        </div>
        <div
          className="sub-title"
        >
          Enter your school-issued student ID number *
        </div>
        <div
          className="inputs"
        >
          <span
            className="student-id-icon"
          />
          <input
            autoFocus={true}
            onChange={[Function]}
            placeholder="School issued ID"
          />
        </div>
        <div
          className="required"
        >
          * required for course credit
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-success btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Continue
        </button>
        <button
          className="cancel btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add it later
        </button>
      </div>
    </div>
  </div>
</div>
`;
